library/src/main/java/com/getstream/sdk/chat/notifications/NotificationsManager.java

```diff
@@ -0,0 +1,26 @@
+package com.getstream.sdk.chat.notifications;
+
+import android.app.Notification;
+import android.content.Context;
+
+import com.getstream.sdk.chat.model.Event;
+import com.getstream.sdk.chat.notifications.options.NotificationOptions;
+import com.google.firebase.messaging.RemoteMessage;
+
+import org.jetbrains.annotations.NotNull;
+
+/*
+ * Created by Anton Bevza on 2019-11-14.
+ */
+public interface NotificationsManager {
+
+    void setFirebaseToken(@NotNull String firebaseToken, @NotNull Context context);
+
+    void onReceiveFirebaseMessage(@NotNull RemoteMessage remoteMessage, @NotNull Context context);
+
+    void onReceiveWebSocketEvent(@NotNull Event event, @NotNull Context context);
+
+    void showNotification(@NotNull Notification notification, @NotNull Context context);
+
+    NotificationOptions getNotificationsOptions();
```

Comment:
Also don't see a reason to expose it for now. Developers can set it when they're creating `NotificationManager`:
```kotlin
StreamChat.setNoticationManager(new NotificationManager(new NotificationOptions()));
StreamChat.init()
```

Reply:
yeah, but in this realization we have to pass NotificationManager before initialize StreamChat. 
And now we init StreamChat in Application. I think adding extra logic will not so good.
In this way we can get default options and add NotificationIntentProvider.

Reply:
That's ok to configure object before using it and init will be hidden anyway, so that the future plan:
```java
val chat = StreamChat.Builder()
  .notifications(CustomNotificationManager))
  .storage(SecureStorage()
  .logger(DebugLogger()
  .style(ChatStle())
  .build()
```
And if we expose `NotificationOptions` some developers might need to fix it.
More over `NotificationOptions` is mutable, that might be another problem. 

The less api methods we expose, the easier it is for support and extension.